.evergreen/download-mongodb.sh

```diff
@@ -699,13 +699,9 @@ download_and_extract_package ()
    MONGODB_DOWNLOAD_URL=$1
    EXTRACT=$2
 
-   if [ -n "$MONGODB_BINARY_ROOT" ]; then
-      cd $MONGODB_BINARY_ROOT
-   else
-      SCRIPT_DIR=$(dirname ${BASH_SOURCE:-$0})
-      . $SCRIPT_DIR/handle-paths.sh
-      cd $DRIVERS_TOOLS
-   fi
+   cd "$(dirname "$(dirname "${MONGODB_BINARIES:?}")")"
+   rm -rf mongodb
```

Comment:
```suggestion
```
This appears to [break expectations](https://parsley.mongodb.com/evergreen/mongo_c_driver_versioned_api_ubuntu2004_test_versioned_api_8.0_patch_eae52c0f3d30be7bbb0029eea878ede649253e02_66ce15e4b8ad300007e7d2a7_24_08_27_18_07_34/0/task?bookmarks=0,387&selectedLineRange=L345) in some existing test code. Can this be removed (given it wasn't present prior)?